### PR TITLE
Added Documentation for Data Structures

### PR DESCRIPTION
**Pull request by Saatvik21**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

`Yes`

### Description

I've added the docs for "What are data structures"

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes : #16

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Kindly have a look at this plagiarism report and fix the plagiarism issues.

[Plagiarism Report](https://github.com/TesseractCoding/NeoAlgo-Docs/files/6283923/smallseotools-1617953420.pdf)

**Comment by Saatvik21** [comment]
> Kindly have a look at this plagiarism report and fix the plagiarism issues.
> 
> [Plagiarism Report](https://github.com/TesseractCoding/NeoAlgo-Docs/files/6283923/smallseotools-1617953420.pdf)

Sorry for the inconvenience, I've updated my documentation. Do let me know if there's still some plagiarism.

File: docs/data-structures/data-structures.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: What are Data Structures?
+slug: /data-structures
+---
+
+## Introduction
+
+A program is a set of instructions that performs operations on data. The way of organizing this data in the main memory at the time of execution of the program is called ***data structures*** .
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:8]: `is called as data structures`

File: docs/data-structures/data-structures.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Depending on the requirement of a program and the type of procedures it is performing, we have various data structures. Some of the common examples of data structures include arrays, linked lists, stacks,  queues, trees and graphs. Data structures
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:10]: `Depending upon the requirement of a program....`

Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:10]: Remove the space before queues. There are two

File: docs/data-structures/data-structures.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+are widely applied in the following areas:
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:11]: Keep this in the previous line itself

File: docs/data-structures/data-structures.mdx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Data structures are classified into two classes: 
+
+#### ***Primitive and Non-primitive Data Structures*** 
+
+Primitive data structures are the basic data types of any programming
+language that form the basic unit for the data structures. Some basic data types are integer, float, character and boolean.           
+    Non-primitive data structures are complex structures that are created using primitive data structures. Examples of such data structures include Arrays, Linked Lists, Stacks, Queues, Trees, and Graphs.
+
+![Flowchart](https://i.imgur.com/0hLAz15.png)
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:29]: Have you made the diagram yourself?

File: docs/data-structures/data-structures.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+#### ***Primitive and Non-primitive Data Structures*** 
+
+Primitive data structures are the basic data types of any programming
+language that form the basic unit for the data structures. Some basic data types are integer, float, character and boolean.           
+    Non-primitive data structures are complex structures that are created using primitive data structures. Examples of such data structures include Arrays, Linked Lists, Stacks, Queues, Trees, and Graphs.
+
+![Flowchart](https://i.imgur.com/0hLAz15.png) 
+<p align = "center"><strong>Fig. 1: Classification of Data Structures</strong></p>
+ 
+
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:32]: Keep only one line break

File: docs/data-structures/data-structures.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+* DBMS
+* Simulation
+* Artificial intelligence 
+* Graphics
+
+## Classification of Data Structures
+
+Data structures are classified into two classes: 
+
+#### ***Primitive and Non-primitive Data Structures***
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:23]: Keep it Heading 3

File: docs/data-structures/data-structures.mdx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+language that form the basic unit for the data structures. Some basic data types are integer, float, character and boolean.           
+    Non-primitive data structures are complex structures that are created using primitive data structures. Examples of such data structures include Arrays, Linked Lists, Stacks, Queues, Trees, and Graphs.
+
+![Flowchart](https://i.imgur.com/0hLAz15.png) 
+<p align = "center"><strong>Fig. 1: Classification of Data Structures</strong></p>
+ 
+
+ Based on the above structure and arrangement of data, non-primitive data structures are further classified into linear and non-linear.
+
+ #### ***Linear and Non-linear Structures***
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:35]: Keep it Heading 3

File: docs/data-structures/data-structures.mdx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+Linear data structures are those where data can be processed in a linear fashion i.e one by one sequentially. It includes the following types of data structures: Arrays, Linked Lists, Stacks & Queues.     
+    Non-linear structures are those in which insertion and deletion are not possible in a linear fashion. It includes the following types of data structures: Trees & Graphs.
+
+## Major Operations in Data Structures
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:39]: `Major Operations on Data Structures`

File: docs/data-structures/data-structures.mdx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+This section discusses the different operations that can be performed on various data structures.
+
+* ***Traversing : *** 
+Accessing each record or element from a given file exactly once in order to process it is called traversing. For
+example, to print the marks of all the students in a class.
+
+* ***Searching : ***
+Finding the location of a given key value in a collection of various data items.
+It can be possible that the given key-value may or may not be present in that collection. For example,
+to find the names of all the people who live in a particular house.
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:50]: Keep them in one line

File: docs/data-structures/data-structures.mdx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+Linear data structures are those where data can be processed in a linear fashion i.e one by one sequentially. It includes the following types of data structures: Arrays, Linked Lists, Stacks & Queues.     
+    Non-linear structures are those in which insertion and deletion are not possible in a linear fashion. It includes the following types of data structures: Trees & Graphs.
+
+## Major Operations in Data Structures
+
+This section discusses the different operations that can be performed on various data structures.
+
+* ***Traversing : *** 
+Accessing each record or element from a given file exactly once in order to process it is called traversing. For
+example, to print the marks of all the students in a class.
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:45]: Keep them in one line

File: docs/data-structures/data-structures.mdx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+* ***Deleting : ***
+Removing the elements from a collection of various data items. For example, to delete the data of a student who has dropped out of a school/college.
+
+* ***Sorting : *** 
+Arranging the elements in the file or structure either in some logical order (alphabetically increasing order, decreasing order) or mathematical order. For example, arranging a deck of cards in ascending or descending order.
+
+* ***Merging : ***
+Combining the elements of two similar data structures to a new data structure of the same type.
+For example, when two lists `List A` and `List B` of size `P` and `Q` respectively, of similar type of elements, are joined to produce the third list, `List C` of size `(P+Q)`, then this process is called merging.
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:63]: Keep them in one line

File: docs/data-structures/data-structures.mdx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+## Advantages of Data Structures
+
+Given below are some important ***advantages of data structures***: 
+
+* Proper choice of data structures makes the program efficient in terms of space and time.
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:69]: `space and time complexity`

File: docs/data-structures/data-structures.mdx [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+* It provides effective and efficient processing of small as well as a large amount of data.
+* They are re-usable.
+* They make the code cleaner and easier to understand.
+* Data structures are specified by an ADT(Abstract Data Type) which provides a level of abstraction.
```
Comment (HarshCasper) [inline docs/data-structures/data-structures.mdx:73]: Give a space after ADT

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Pretty neat job. Few changes here ⬇️

File: docs/data-structures/data-structures.mdx (hunk 4 above)
Comment (Saatvik21) [reply docs/data-structures/data-structures.mdx:29]: Yes I made it using Creately app. Here's the [link](https://app.creately.com/diagram/HCKrmS4ayh6/view) for the same.

**Comment by Saatvik21** [comment]
> Pretty neat job. Few changes here ⬇️

Done!

**Review by nidh-ish — APPROVED** [review]

**Pull request merged**